### ISSUE TITLE
Alpha not working on images

### ISSUE THREAD

**Issue by nerduno** [bokeh/bokeh#7474]
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
bokeh: 0.12.10
python: 2.7
notebook: 5.0.0
jupyter: 1.0.0
jupyter_client 5.1.0
jupyter_console: 5.2.0
jupyter_core: 4.3.0
ipython: 5.4.1
ipython_genutils: 0.2.0
OS: ubuntu 14.04
broswer: Safari Version 11.0.1 (13604.3.5)

#### Description of expected behavior and the observed behavior
Setting the alpha of an image should change its transparency, but setting alpha has no effect (I tried setting to 0,.5,1,128,255).

#### Complete, minimal, self-contained example code that reproduces the issue
```
from skimage import data
image = data.coins() #Any image will do, but I didn't find a sample image in bokeh's sample data
fig_overlay = figure(x_range=(0, image.shape[1]), y_range=(0, image.shape[0]),
                     plot_width=image.shape[1]*1, plot_height=image.shape[0]*1)
bkimg1 = fig_overlay.image(image=[image], x=0, y=0, dw=image.shape[1], dh=image.shape[0], alpha=.5) 
bkimg2 = fig_overlay.image(image=[image], x=100, y=100, dw=image.shape[1], dh=image.shape[0], alpha=.5) #changing alpha has no effect on the resulting figure.
show(fig_overlay)
```

**Comment by bryevdv** [comment]
As things currently stand, this really ought to throw an exception. `alpha` is a shortcut for setting `line_alpha` and `fill_alpha` at the same time, neither of which `Image` has. 

I'm not sure about adding a global alpha, either. I think the previous consensus was that the color mapper would be the place to apply an alpha.

**Comment by nerduno** [comment]
Thanks for clarifying. I think adding a comment to the documentation for the figure.image method would be sufficient.

**Comment by marcuskd** [comment]
I would also find alpha in images very useful. I'm happy to have a go at a PR if that would be helpful?

**Comment by bryevdv** [comment]
@marcuskd that would be very helpful. The other image glyph with `global_alpha` should be a good template but let me know any questions

**Comment by marcuskd** [comment]
Thanks Bryan, will do.

**Comment by bryevdv** [comment]
closed by #7570
